.github/workflows/rust.yml

```diff
@@ -39,9 +38,6 @@ jobs:
           rustup component add clippy
           rustup toolchain install $NIGHTLY --allow-downgrade -c rustfmt --no-self-update
 
-      - name: Generate Cargo.lock
-        run: cargo +${{ env.NIGHTLY }} -Z minimal-versions update
```

Comment:
Hmm why are we removing this?

Reply:
This line goal is to ensure that the minimum version specified in the Cargo.toml compiles, it's not for finding appropriate version but for testing our Cargo.toml dependency requirements are correct

Reply:
That makes sense, I'll add it back.